lib/azure/armrest/storage/managed_disk.rb

```diff
@@ -0,0 +1,29 @@
+# Azure namespace
+module Azure
+  # Armrest namespace
+  module Armrest
+    # Storage namespace
+    module Storage
+      # Base class for managing managed disks.
+      module ManagedStorageHelper
+        class ManagedDisk
+          def initialize(storage_service, disk_name, resource_group, sas_url)
+            @storage_service = storage_service
+            @disk_name       = disk_name
+            @resource_group  = resource_group
+            @sas_url         = sas_url
```

Comment:
Can't you acquire the `sas_url` here and not have to pass it in? This way you encapsulate the `ManagedDisk` semantics in this class. Then, any needed re-acquisition of the `sas_url` can happen here as well - in the case of a failed `read`.

Reply:
Oh, how I tried.  For two days. I could not for the life of me figure out how to get the ManagedDisk work with the modules in question and allow it to call all the methods that live up in AzureArmrestService that are required to do everything necessary.  If somebody wants to show me how to do that I am more than willing to change this all again.

Reply:
What's missing in `ManagedDisk`? You have:
```ruby
def initialize(storage_service, disk_name, resource_group, sas_url)
```
passed in to the initializer.
In the instantiating code you call:
```ruby
sas_url = access_token(disk_name, resource_group, options)
```
So you have: `disk_name` and `resource_group` needed to call `access_token()` from `storage_service`.
It seems like all you need to do is pass `options` into initialize as well.